file/lib/src/sentry_file_extension.dart

```diff
@@ -0,0 +1,42 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:io' if (dart.library.html) 'dart:html';
+
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+
+import '../sentry_file.dart';
+
+extension SentryFileExtension on File {
+  /// The Sentry wrapper for the File IO implementation that creates a span
+  /// out of the active transaction in the scope.
+  /// The span is started before the operation is executed and finished after.
+  /// The File tracing isn't available for Web.
+  ///
+  /// Example:
+  ///
+  /// ```dart
+  /// import 'dart:io';
+  ///
+  /// final file = File('test.txt');
+  /// final sentryFile = SentryFile(file);
+  /// // span starts
+  /// await sentryFile.writeAsString('Hello World');
+  /// // span finishes
+  /// ```
+  ///
+  /// All the copy, create, delete, open, rename, read, and write operations are
+  /// supported.
+  File sentryTrace({
```

Comment:
What's the reason to not just automatically enable it for all file io?

You can do that with the following code:
```dart
IOOverrides.global = SentryIoOverrides(HubAdapter(), options);

File f = File('path/to/file'); // file is SentryFile by default

class SentryIoOverrides extends IOOverrides {
  final Hub _hub;
  final SentryOptions _options;

  SentryIoOverrides(this._hub, this._options);

  @override
  File createFile(String path) {
    if (!_options.isTracingEnabled()) {
      return super.createFile(path);
    }
    return SentryFile(
      super.createFile(path),
      _hub,
      _options,
    );
  }
}
```

Reply:
[I did see that](https://api.dart.dev/stable/2.18.5/dart-io/IOOverrides-class.html) and it is indeed awesome, but its also misusing the mocks API, as an SDK, we need to be a bit more conservative about that, I guess.
I'm afraid that this could break the app's behavior that relies on the `File` type or something like that, or, that some files take more time than usual and you don't want to trace them for some reason (overhead, long transactions, etc), an all in solution could be a deal breaker for some users.
I'd prefer to do this next after gathering some feedback on the current implementation.
What do you think?

Reply:
Yeah, that sounds reasonable

Reply:
I was mostly asking because `IOOverrides` allow you to also measure the performance, if you don't otherwise control the file io yourself. Like for example with [Hive](https://pub.dev/packages/hive), [sembast](https://pub.dev/packages/sembast) or similar.

Reply:
Yep, I believe it still makes sense to have it, its a hook that makes it possible since Dart does not have reflection nor monkey patching, not sure if it'd be ever enabled by default tho.
At least not during the first version.